src/unit_tests/logcollector/test_logcollector.c

```diff
@@ -40,119 +40,173 @@ void w_initialize_file_status();
 int w_update_hash_node(char * path, int64_t pos);
 int w_set_to_last_line_read(logreader *lf);
 
+// Auxiliar structs
+typedef struct test_logcollector_s {
+    logreader *log_reader;
+    SHA_CTX *context;
+    os_file_status_t *status;
+} test_logcollector_t;
 
 /* setup/teardown */
 
 static int setup_group(void **state) {
     test_mode = 1;
     return 0;
 }
+
 static int teardown_group(void **state) {
     test_mode = 0;
     return 0;
 }
 
+static int setup_hashmap(void **state) {
+    files_status = OSHash_Create();
+    if (files_status == NULL) {
+        return 1;
+    }
+
+    OSHash_SetFreeDataPointer(files_status, free);
+
+    return 0;
+}
+
+static int teardown_hashmap(void **state) {
+    OSHash_Free(files_status);
+    return 0;
+}
+
+static int setup_log_context(void **state) {
+    if (setup_hashmap(state) != 0) {
+        return 1;
+    }
+
+    test_logcollector_t *test_struct = calloc(1, sizeof(test_logcollector_t));
+    if (test_struct == NULL) {
+        return 1;
+    }
+
+    test_struct->log_reader = calloc(1, sizeof(logreader));
+    test_struct->context = calloc(1, sizeof(SHA_CTX));
+    test_struct->status = calloc(1, sizeof(os_file_status_t));
+    if (test_struct->log_reader == NULL || test_struct->context == NULL || test_struct->status == NULL) {
+        return 1;
+    }
+
+    test_struct->log_reader->fp = (FILE *) 1;
+    *state = test_struct;
+    return 0;
+}
+
+static int teardown_log_context(void **state) {
+    teardown_hashmap(state);
```

Comment:
We should check the return value of this teardown and fail ourselves if it fails. We might also want to move it down so if it fails the rest of the memory is correctly free'd